server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/EventDeadLettersIntegrationTest.java

```diff
@@ -0,0 +1,473 @@
+        retryEventsListener.setRetriesBeforeSuccess(3);
+        generateInitialEvent();
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups")
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body(".", hasSize(0));
+    }
+
+    @Test
+    public void failedEventGroupShouldBeStoredInDeadLetter() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        generateInitialEvent();
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups")
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body(".", containsInAnyOrder(SERIALIZED_GROUP));
+    }
+
+    @Test
+    public void failedEventShouldBeStoredInDeadLetter() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        generateInitialEvent();
+
+        String failedEventId = retrieveFirstFailedEventId();
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP + "/" + failedEventId)
+        .then()
+            .statusCode(HttpStatus.OK_200);
+    }
+
+    @Test
+    public void multipleFailedEventShouldBeStoredInDeadLetter() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        generateInitialEvent();
+        generateSecondEvent();
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP)
+        .then()
+            .statusCode(HttpStatus.OK_200)
+            .contentType(ContentType.JSON)
+            .body(".", hasSize(2));
+    }
+
+    @Test
+    public void failedEventShouldNotBeInDeadLetterAfterBeingDeleted() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        generateInitialEvent();
+
+        String failedEventId = retrieveFirstFailedEventId();
+
+        when()
+            .delete(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP + "/" + failedEventId)
+        .then()
+            .statusCode(HttpStatus.NO_CONTENT_204);
+
+        when()
+            .get(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP + "/" + failedEventId)
+        .then()
+            .statusCode(HttpStatus.NOT_FOUND_404);
+    }
+
+    @Test
+    public void failedEventShouldNotBeInDeadLettersAfterSuccessfulRedelivery() {
+        retryEventsListener.setRetriesBeforeSuccess(4);
+        generateInitialEvent();
+
+        String failedEventId = retrieveFirstFailedEventId();
+
+        String taskId = with()
+            .queryParam("action", EVENTS_ACTION)
+        .post(EventDeadLettersRoutes.BASE_PATH + "/groups/" + SERIALIZED_GROUP + "/" + failedEventId)
+            .jsonPath()
+            .get("taskId");
+
+        given()
+            .basePath(TasksRoutes.BASE)
+        .when()
+            .get(taskId + "/await")
+        .then()
+            .body("status", is("completed"))
```

Comment:
either you check the task status or the fact that the event is not in dead letter anymore but not both.